osutils.pm

```diff
@@ -105,30 +106,80 @@ sub runcmd {
 
     my ($wtr, $rdr, $err);
     $err = gensym;
-    $pid = open3($wtr, $rdr, $err, @_);
+    $pid = open3($wtr, $rdr, $err, @$params);
     die "couldn't open: $!" unless defined $pid;
     close($wtr) or die "couldn't close fh: $!";
 
-    my $s = IO::Select->new();
-    $s->add($rdr, $err);
-    while (my @ready = $s->can_read()) {
-        for my $fh (@ready) {
-            if (sysread($fh, my $buf, 4096)) {
-                diag $buf if ($fh == $rdr);
-                diag $buf if ($fh == $err);
-            }
-            else {
-                $s->remove($fh);
-            }
-        }
-    }
+    $sub->($rdr, $err);
+
     close($rdr) or die "couldn't close fh: $!";
     close($err) or die "couldn't close fh: $!";
 
     my $exit_code = $status >> 8;
```

Comment:
Strangely I have now this message in my logs that was not here before: "Use of uninitialized value $status in right bitshift (>>) at /usr/lib/os-autoinst/osutils.pm line 118."

I don't know if my test is failing because of this (http://1b147.qa.suse.de/tests/2681/file/autoinst-log.txt)...

Reply:
FYI https://progress.opensuse.org/issues/37580